core/src/main/java/io/micronaut/core/value/PropertyResolver.java

```diff
@@ -89,6 +89,29 @@ public interface PropertyResolver extends ValueResolver<String> {
         return Collections.emptySet();
     }
 
+    /**
+     * Returns a collection of properties entries under the given key, but .
+     * For example, if you set {@code PropertyCatalog.RAW} then the following keys:
+     *
+     * <pre>
+     * <code>datasource.MyDs-1.url=localhost
+     * datasource.MyDs-2.url=someother</code>
+     * </pre>
+     *
+     * Calling {@code getPropertyEntries(String)} with a value of {@code datasource} will result in a collection
+     * containing {@code MyDs-1} and {@code MyDs-2} (without normalization).
+     *
+     * @param name The name to resolve
+     * @param propertyCatalog property catalog to use
+     * @return The property entries.
+     *
+     * @since 4.7.0
+     */
+    @NonNull
+    default Collection<String> getPropertyEntries(@NonNull String name, PropertyCatalog propertyCatalog) {
```

Comment:
add nullability annotation to second parameter

Reply:
fixed